client/src/app/users/user-list.component.ts

```diff
@@ -108,9 +107,8 @@ export class UserListComponent implements OnInit, OnDestroy  {
    * Called when the filtering information is changed in the GUI so we can
    * get an updated list of `filteredUsers`.
    */
-  public updateFilter(): void {
-    this.filteredUsers = this.userService.filterUsers(
-      this.serverFilteredUsers, { name: this.userName, company: this.userCompany });
+  public updateFilter(): User[] {
+     return this.userService.filterUsers(this.serverFilteredUsers, { name: this.userName, company: this.userCompany });
```

Comment:
This is the problem. The HTML for this component calls `updateFilter()` whenever there is a change in the input fields. It then expects that to change the value of `this.filteredUsers`, leading to a change in the displayed users. By making this more functional, we lost that expected side-effect, so typing in the text boxes has no effect.

If we change this to:

```typescript
  public updateFilter(): User[] {
     this.filteredUsers = this.userService.filterUsers(this.serverFilteredUsers, { name: this.userName, company: this.userCompany });
     return this.filteredUsers;
  }
```
everything works again.

This does raise the question, though, of whether there's a way to avoid this use of state/side-effects. We could have the HTML call `userService.filterUsers()` every time it wants a filtered list of users?

Reply:
Ah, yeah I see the issue now. Sorry, I would have realized this quick if I could run it. Having issues getting Mongo working with my WSL configuration.

I can make that change above to get it working if you want. 

Otherwise, check out this commit in the alt branch: https://github.com/UMM-CSci-3601/3601-iteration-template/commit/8f636136b87bf87eff3edceca79789398ef40976

`filteredUsers` would then be a `computed` value. Whenever the `serverFilteredUsers` changes or `userName` or `userCompany` changes, the filter would fire off and update the computed value.

Reply:
I pushed up your suggested change

Reply:
The signal ngModel binding in that linked branch is now supported, and those relevant signals firing would avoid the need for calling a function like this new version that has side effects

Reply:
This all said, not sure if that would actually work without getting Mongo running. If that approach did work, IMO that is the best way to not have side effects.